CMakeLists.txt

```diff
@@ -30,6 +30,19 @@ FetchContent_Declare(
   GIT_TAG        f9ff166845a59327eda431af82ee85a9c7532c5d
 )
 
+if(NOT WIN32)
+  if(EXISTS ${CMAKE_SYSROOT}/usr/lib/x86_64-linux-gnu/libvulkan.so)
+    set(Vulkan_INCLUDE_DIR ${CMAKE_SYSROOT}/usr/include)
+    set(Vulkan_LIBRARY ${CMAKE_SYSROOT}/usr/lib/x86_64-linux-gnu/libvulkan.so)
+  endif()
+endif()
+find_package(Vulkan)
+
+if(Vulkan_FOUND)
+    set(SD_FLASH_ATTN ON)
+    set(SD_VULKAN ON)
+endif()
+
 set(IMPLIB_SOURCE_FILES "")
 if (CUDAToolkit_FOUND)
   set(SD_CUBLAS ON)

```

Comment:
do we need to check !Vulkan_FOUND to disable the CUDA paths here? I don't think one binary can support both in GGML currently.

Same for all the if (CUDAToolkit_FOUND)s

Reply:
You're right its either/or. I updated it like you suggested, so it will build with Vulkan if its installed and disable CUDA in that case.